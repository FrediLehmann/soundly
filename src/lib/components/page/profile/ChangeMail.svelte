<script lang="ts" context="module">
  import { Button, Flyin, Input } from '$lib/components/atoms';
  import { supabase } from '$lib/supabase';
</script>

<script lang="ts">
  export let email: string;

  let flyin: Flyin;
  let emailError = '';

  let editMode = false;
  const toggleEdit = () => (editMode = !editMode);

  const saveNewEmail = async () => {
    const emailInput: HTMLFormElement =
      document.querySelector('input[id="email"]');

    if (!emailInput.validity.valid) {
      emailError = emailInput.validationMessage;
      return;
    } else {
      emailError = '';
    }

    const { error } = await supabase.auth.update({ email });
    if (error) {
      flyin.show('New email address could not be saved.', 'error');
    } else {
      flyin.show(
        'Email address successfull changed. Please verify your new email address.',
        'success'
      );
      editMode = false;
    }
  };
</script>

<section>
  <h2>Email</h2>
  <p>
    Your email address is used to identify you, such as on login but it is <b
      >not public</b
    >. You can change your email address whenever, but it will trigger an email
    confirmation process.
  </p>
  <div class="input">
    <Input
      name="email"
      label="Email"
      type="email"
      required
      disabled={!editMode}
      error={emailError}
      labelSrOnly={true}
      bind:value={email} />
  </div>
  {#if editMode}
    <Button on:click={toggleEdit}>Cancel</Button>
    <Button style="primary" on:click={saveNewEmail}>Save new Email</Button>
  {:else}
    <Button on:click={toggleEdit}>Change Email</Button>
  {/if}
</section>

<Flyin bind:this={flyin} />

<style>
  h2 {
    font-size: 1.2rem;
  }

  .input {
    max-width: 70%;
    margin-block-end: 1rem;
  }

  section {
    max-width: 75%;
  }
</style>
