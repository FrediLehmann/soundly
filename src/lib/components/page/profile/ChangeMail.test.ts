/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom';
import { fireEvent, render } from '@testing-library/svelte';

import ChangeMail from './ChangeMail.svelte';

describe('<ChangeMail /> component', () => {
  test('renders correctly', () => {
    const { component } = render(ChangeMail);
    expect(component).toMatchSnapshot();
  });

  test('renders edit mode correctly', async () => {
    const { component, getByText } = render(ChangeMail);
    await fireEvent.click(getByText('Change Email'));
    expect(component).toMatchSnapshot();
  });
});
