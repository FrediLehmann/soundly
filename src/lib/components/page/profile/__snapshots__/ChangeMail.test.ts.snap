// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChangeMail /> component renders correctly 1`] = `
ChangeMail {
  "$$": Object {
    "after_update": Array [],
    "before_update": Array [],
    "bound": Object {},
    "callbacks": Object {},
    "context": Map {},
    "ctx": Array [
      "",
      Flyin {
        "$$": Object {
          "after_update": Array [],
          "before_update": Array [],
          "bound": Object {},
          "callbacks": Object {},
          "context": Map {},
          "ctx": Array [
            false,
            undefined,
            undefined,
            [Function],
            [Function],
          ],
          "dirty": Array [
            -1,
          ],
          "fragment": Object {
            "c": [Function],
            "d": [Function],
            "i": [Function],
            "l": [Function],
            "m": [Function],
            "o": [Function],
            "p": [Function],
          },
          "not_equal": [Function],
          "on_destroy": Array [],
          "on_disconnect": Array [],
          "on_mount": Array [],
          "props": Object {
            "show": 4,
          },
          "root": <div>
            <section
              class="svelte-1sz96f6"
            >
              <h2
                class="svelte-1sz96f6"
              >
                Email
              </h2>
               
              <p>
                Your email address is used to identify you, such as on login but it is 
                <b>
                  not public
                </b>
                . You can change your email address whenever, but it will trigger an email
    confirmation process.
              </p>
               
              <div
                class="input svelte-1sz96f6"
              >
                <div
                  class="container svelte-1ulq3bj"
                  data-required="true"
                  data-sr-only="true"
                >
                  <label
                    class="svelte-1ulq3bj"
                    for="email"
                  >
                    Email
                  </label>
                   
                  <input
                    class="svelte-1ulq3bj"
                    disabled=""
                    id="email"
                    required=""
                    type="email"
                  />
                   
                </div>
              </div>
               
              <button
                class="svelte-1dsjpe6"
                data-type="secondary"
                type="button"
              >
                Change Email
              </button>
              
            </section>
             
            
          </div>,
          "skip_bound": false,
          "update": [Function],
        },
        "$capture_state": [Function],
        "$inject_state": [Function],
      },
      "",
      false,
      [Function],
      [Function],
      [Function],
      [Function],
    ],
    "dirty": Array [
      -1,
    ],
    "fragment": Object {
      "c": [Function],
      "d": [Function],
      "i": [Function],
      "l": [Function],
      "m": [Function],
      "o": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": Array [
      [Function],
    ],
    "on_disconnect": Array [],
    "on_mount": Array [],
    "props": Object {
      "email": 0,
    },
    "root": <div>
      <section
        class="svelte-1sz96f6"
      >
        <h2
          class="svelte-1sz96f6"
        >
          Email
        </h2>
         
        <p>
          Your email address is used to identify you, such as on login but it is 
          <b>
            not public
          </b>
          . You can change your email address whenever, but it will trigger an email
    confirmation process.
        </p>
         
        <div
          class="input svelte-1sz96f6"
        >
          <div
            class="container svelte-1ulq3bj"
            data-required="true"
            data-sr-only="true"
          >
            <label
              class="svelte-1ulq3bj"
              for="email"
            >
              Email
            </label>
             
            <input
              class="svelte-1ulq3bj"
              disabled=""
              id="email"
              required=""
              type="email"
            />
             
          </div>
        </div>
         
        <button
          class="svelte-1dsjpe6"
          data-type="secondary"
          type="button"
        >
          Change Email
        </button>
        
      </section>
       
      
    </div>,
    "skip_bound": false,
    "update": [Function],
  },
  "$$set": [Function],
  "$capture_state": [Function],
  "$inject_state": [Function],
}
`;

exports[`<ChangeMail /> component renders edit mode correctly 1`] = `
ChangeMail {
  "$$": Object {
    "after_update": Array [],
    "before_update": Array [],
    "bound": Object {},
    "callbacks": Object {},
    "context": Map {},
    "ctx": Array [
      "",
      Flyin {
        "$$": Object {
          "after_update": Array [],
          "before_update": Array [],
          "bound": Object {},
          "callbacks": Object {},
          "context": Map {},
          "ctx": Array [
            false,
            undefined,
            undefined,
            [Function],
            [Function],
          ],
          "dirty": Array [
            -1,
          ],
          "fragment": Object {
            "c": [Function],
            "d": [Function],
            "i": [Function],
            "l": [Function],
            "m": [Function],
            "o": [Function],
            "p": [Function],
          },
          "not_equal": [Function],
          "on_destroy": Array [],
          "on_disconnect": Array [],
          "on_mount": Array [],
          "props": Object {
            "show": 4,
          },
          "root": <div>
            <section
              class="svelte-1sz96f6"
            >
              <h2
                class="svelte-1sz96f6"
              >
                Email
              </h2>
               
              <p>
                Your email address is used to identify you, such as on login but it is 
                <b>
                  not public
                </b>
                . You can change your email address whenever, but it will trigger an email
    confirmation process.
              </p>
               
              <div
                class="input svelte-1sz96f6"
              >
                <div
                  class="container svelte-1ulq3bj"
                  data-required="true"
                  data-sr-only="true"
                >
                  <label
                    class="svelte-1ulq3bj"
                    for="email"
                  >
                    Email
                  </label>
                   
                  <input
                    class="svelte-1ulq3bj"
                    id="email"
                    required=""
                    type="email"
                  />
                   
                </div>
              </div>
               
              <button
                class="svelte-1dsjpe6"
                data-type="secondary"
                type="button"
              >
                Cancel
              </button>
              
               
              <button
                class="svelte-1dsjpe6"
                data-type="primary"
                type="button"
              >
                Save new Email
              </button>
              
            </section>
             
            
          </div>,
          "skip_bound": false,
          "update": [Function],
        },
        "$capture_state": [Function],
        "$inject_state": [Function],
      },
      "",
      true,
      [Function],
      [Function],
      [Function],
      [Function],
    ],
    "dirty": Array [
      -1,
    ],
    "fragment": Object {
      "c": [Function],
      "d": [Function],
      "i": [Function],
      "l": [Function],
      "m": [Function],
      "o": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": Array [
      [Function],
    ],
    "on_disconnect": Array [],
    "on_mount": Array [],
    "props": Object {
      "email": 0,
    },
    "root": <div>
      <section
        class="svelte-1sz96f6"
      >
        <h2
          class="svelte-1sz96f6"
        >
          Email
        </h2>
         
        <p>
          Your email address is used to identify you, such as on login but it is 
          <b>
            not public
          </b>
          . You can change your email address whenever, but it will trigger an email
    confirmation process.
        </p>
         
        <div
          class="input svelte-1sz96f6"
        >
          <div
            class="container svelte-1ulq3bj"
            data-required="true"
            data-sr-only="true"
          >
            <label
              class="svelte-1ulq3bj"
              for="email"
            >
              Email
            </label>
             
            <input
              class="svelte-1ulq3bj"
              id="email"
              required=""
              type="email"
            />
             
          </div>
        </div>
         
        <button
          class="svelte-1dsjpe6"
          data-type="secondary"
          type="button"
        >
          Cancel
        </button>
        
         
        <button
          class="svelte-1dsjpe6"
          data-type="primary"
          type="button"
        >
          Save new Email
        </button>
        
      </section>
       
      
    </div>,
    "skip_bound": false,
    "update": [Function],
  },
  "$$set": [Function],
  "$capture_state": [Function],
  "$inject_state": [Function],
}
`;
